### PR TITLE
lint(engine): fix all lint errors in engine module

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve?

Issue Number: ref #7271

### What is changed and how it works?

- Fix all errors under the engine module, with `golangci-lint v1.50.0`

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by lance6716 — APPROVED** [review]

File: engine/test/e2e/e2e_dm_test.go [hunk 1]
```diff
@@ -515,7 +517,7 @@ func getBinlogSchema(client *http.Client, jobID string, task string, schema stri
 		v.Add("table", table)
 	}
 	u += "?" + v.Encode()
-	resp, err := client.Get(u)
+	resp, err := client.Get(ctx, u) //nolint:bodyclose
```
Comment (gozssky) [inline engine/test/e2e/e2e_dm_test.go:520]: We can close `resp.Body` after error check.

File: pkg/httputil/httputil_test.go [hunk 2]
```diff
@@ -70,25 +70,55 @@ func TestHttputilNewClient(t *testing.T) {
 	require.Equal(t, httputilServerMsg, string(body))
 }
 
+func TestStatusCodeCreated(t *testing.T) {
+	t.Parallel()
+
+	port := 8305
```
Comment (gozssky) [inline pkg/httputil/httputil_test.go:76]: It seems we can use `httptest.NewServer` to avoid port conflict.

**Review by gozssky — APPROVED** [review]
Rest LGTM.

File: pkg/httputil/httputil_test.go (hunk 2 above)
Comment (amyangfei) [reply pkg/httputil/httputil_test.go]: addressed.

**Pull request merged**